openwisp_users/multitenancy.py

```diff
@@ -134,3 +136,52 @@ class MultitenantRelatedOrgFilter(MultitenantOrgFilter):
     """
 
     multitenant_lookup = 'organization__in'
+
+
+class BaseOrganizationPermission(BasePermission):
+    def get_organization(self, view, obj):
+        organization_field = getattr(view, 'organization_field', 'organization')
+        return getattr(obj, organization_field, None)
+
+    def validate_membership(self, user, org):
+        raise NotImplementedError(
+            'Please use one of the child classes: IsOrganizationMember,'
+            'IsOrganizationManager or IsOrganizationOwner'
+        )
+
+    def has_permission(self, request, view):
+        return request.user and request.user.is_authenticated
+
+    def has_object_permission(self, request, view, obj):
```

Comment:
this method should be defined first to improve readability, then has_permission, get_object_organization and validate_membership last.